client/my-sites/plugins/search-box-header/index.jsx

```diff
@@ -20,8 +20,9 @@ const SearchBox = ( { isMobile, doSearch, searchTerm, searchBoxRef } ) => {
 				fitsContainer={ isMobile }
 				onSearch={ doSearch }
 				defaultValue={ searchTerm }
+				searchMode="on-enter"
 				placeholder={ translate( 'Try searching "ecommerce"' ) }
-				delaySearch={ true }
+				delaySearch={ false }
```

Comment:
```suggestion
```
I think this might be an option now. I think I changed this thinking I'd use doSearch internally but it made more sense to use onSearch directly for the "on-enter" mode.